controllers/utils/fakedns.go

```diff
@@ -190,7 +200,7 @@ func (m *DNSMock) handleReflect(w dns.ResponseWriter, r *dns.Msg) {
 	if m.records[r.Question[0].Qtype] != nil {
 		for _, rr := range m.records[r.Question[0].Qtype] {
 			fqdn := strings.Split(rr.String(), "\t")[0]
-			if fqdn == r.Question[0].Name {
+			if fqdn == r.Question[0].Name || m.settings.Dump {
```

Comment:
Does it make more sense to make the handler function composable, so each test can use its own handler function instead of doing this?  That might allow more flexibility going forward

Reply:
A custom function would allow more flexibility, but be more intrusive since we would have to define it in more places.
We can keep it like this and reevaluate if new cases arise